runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -351,17 +350,20 @@ public int count() {
     @Nonnull
     List<CorfuStoreEntry<K, V, M>> scanAndFilterByEntry(
             @Nonnull final Predicate<CorfuStoreEntry<K, V, M>> entryPredicate) {
-        return corfuTable.scanAndFilterByEntry(recordEntry ->
-                entryPredicate.test(new CorfuStoreEntry<>(
-                        recordEntry.getKey(),
-                        recordEntry.getValue().getPayload(),
-                        recordEntry.getValue().getMetadata())))
-                .parallelStream()
-                .map(entry -> new CorfuStoreEntry<>(
-                        entry.getKey(),
-                        entry.getValue().getPayload(),
-                        entry.getValue().getMetadata()))
-                .collect(Collectors.toList());
+        try(Stream<Map.Entry<K, CorfuRecord<V, M>>> stream = corfuTable.entryStream()) {
```

Comment:
I think original impl does not have a deep copy that introduced by "ImmutableListSetWrapper.fromMap".

This change will have higher latency caused by the deep copy, right?

Reply:
It's a trade-off between latency vs throughput. The copy only creates a new map "structure", but doesn't do a copy on the key/value. 

```java

public static <K, V> ImmutableListSetWrapper<Entry<K, V>> fromMap(@NonNull Map<K, V> original) {
        return new ImmutableListSetWrapper<>(original
                .entrySet()
                .stream()
                .map(entry -> new SimpleImmutableEntry<>(entry.getKey(), entry.getValue()))
                .collect(ImmutableList.toImmutableList()));
```

The reason this is safe, that is we don't need  a deep copy, is because Protobuf objects are immutable. More often than note (at least based on my observations), making a shallow copy of the map is much cheaper than waiting on the VLO lock.